qiskit/wrapper/_wrapper.py

```diff
@@ -38,17 +37,33 @@ def register(token, url='https://quantumexperience.ng.bluemix.net/api',
             proxies (dict): Proxy configuration for the API, as a dict with
                 'urls' and credential keys.
             verify (bool): If False, ignores SSL certificates errors.
-            provider_name (str): the unique name for the online backend
-                provider (for example, 'ibmq' for the IBM Quantum Experience).
+            provider_name (str): the user name for the registered provider.
     Raises:
         QISKitError: if the provider name is not recognized.
     """
-    if provider_name == 'ibmq':
-        provider = IBMQProvider(token, url,
-                                hub, group, project, proxies, verify)
-        _DEFAULT_PROVIDER.add_provider(provider)
-    else:
-        raise QISKitError('provider name %s is not recognized' % provider_name)
+    # Convert the credentials to a dict.
+    credentials = {
+        'token': token, 'url': url, 'hub': hub, 'group': group,
+        'project': project, 'proxies': proxies, 'verify': verify
+    }
+    _DEFAULT_PROVIDER.add_ibmq_provider(credentials, provider_name)
+
+
+def unregister(provider_name):
+    """
+    Removes a provider of list of registered providers.
+
+    Args:
+        provider_name (str): The unique name for the online provider.
+    Raises:
+        QISKitError: if the provider name is not valid.
+    """
+    _DEFAULT_PROVIDER.remove_provider(provider_name)
```

Comment:
Hmm, I'm unsure if it would play nicely with the philosophy of not having the end-user-oriented wrapper functions expose the providers directly (and in this particular case, would return a "not longer in use provider"). It's a good point, though - perhaps we can reevaluate in the parent PR?